Adding docs for 1.5 toolkit features (#804)
timescale/docs: api/values-freq_agg.md

# values()  <tag type="toolkit">Toolkit</tag><tag type="experimental">Experimental</tag>

> This will return the data accumulated in a [frequency aggregate](hyperfunctions/frequency-analysis/freq_agg/).  Note that since the aggregate operates over `AnyElement` types, this method does require a type parameter to determine the type of the output.

Reviewer: ```suggestion
This function returns the data accumulated in a 
[frequency aggregate](hyperfunctions/frequency-analysis/freq_agg/).  
The aggregate operates over `AnyElement` types, so this method 
requires a type parameter to determine the type of the output.
```

> This API is experimental and is subject to change without notice.

Reviewer: ```suggestion
<highlight type="warning">
Experimental features could have bugs! They might not be backwards compatible,
and could be removed in future releases. Use these features at your own risk, and
do not use any experimental features in production.
</highlight>
```

## Required arguments

> |Name| Type |Description|

Reviewer: ```suggestion
|Name|Type|Description|
```

|-|-|-|
|`agg`|`FrequencyAggregate`|The aggregate to display the values for|
|`ty`|`AnyElement`|A value matching the type to output from the aggregate|

## Returns

|Column|Type|Description|
|-|-|-|
|`value`|`AnyElement`|One of the most common values in the data from which the aggregate was created|
|`min_freq`|`DOUBLE PRECISION`|The minimum frequency of the value in the originating data|
|`max_freq`|`DOUBLE PRECISION`|The maximum frequency of the value in the originating data|

## Sample usage

> This test uses a table of randomly generated data.  The values here will be the integer square roots of random number in the range (0,400).

Reviewer: ```suggestion
This test uses a table of randomly generated data. The values used are the integer 
square roots of a random number in the range (0,400).
```

> 

Reviewer: ```suggestion
```

```sql
CREATE TABLE value_test(value INTEGER);
INSERT INTO value_test SELECT floor(sqrt(random() * 400)) FROM generate_series(1,100000);
```

> The following will now return the values representing more than 5% of the input.

Reviewer: ```suggestion
This returns values representing more than 5% of the input:
```

> 

Reviewer: ```suggestion
```

```sql
SELECT value, min_freq, max_freq
FROM toolkit_experimental.values(
    (SELECT toolkit_experimental.freq_agg(0.05, value) FROM value_test),
    0::INTEGER);
```

> The output for this query (expect some small variation due to randomness if you try to reproduce this):

Reviewer: ```suggestion
The output for this query looks like this, with some variation due to randomness:
```
